lib/internal/child_process.js

```diff
@@ -912,7 +927,15 @@ function setupChannel(target, channel, serializationMode) {
   };
 
   function emit(event, message, handle) {
-    target.emit(event, message, handle);
+    const args = [event, message, handle];
+    const isInternalMessage = "internalMessage" === event;
+    const hasListenersInstalled = target.listenerCount('message');
+    if (hasListenersInstalled || isInternalMessage) { 
+      target.emit(...args);
```

Comment:
```suggestion
      target.emit(event, message, handle);
```

Reply:
Sorry I didn't get why this one. The args was created on a line before so I didn't call all args directly to reuse it

Reply:
fixed!

Reply:
I didn't understand why Array.iterator[Symbol.iterator] is globally mutable, could you give an explanation? (I know this topic is old, but I have this question)

Reply:
That’s how it’s defined in the ECMAScript spec – you can find more information in https://github.com/nodejs/node/blob/18800da280f87ebc8ad60afcbd8ce48fa2127bf6/doc/contributing/primordials.md?plain=1#L262-L270